core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -277,6 +281,60 @@ public final V replace(K key, V value) {
       return replace(key, value, defaultMetadata);
    }
 
+
+   @Override
+   public V compute(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+     return compute(key, remappingFunction, false);
+   }
+
+   @Override
+   public V compute(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction, Metadata metadata) {
+      return computeInternal(key, remappingFunction, false, metadata, addUnsafeFlags(EnumUtil.EMPTY_BIT_SET));
+   }
+
+   @Override
+   public V computeIfPresent(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+      return compute(key, remappingFunction, true);
+   }
+
+   @Override
+   public V computeIfPresent(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction, Metadata metadata) {
+      return computeInternal(key, remappingFunction, true, metadata, addUnsafeFlags(EnumUtil.EMPTY_BIT_SET));
+   }
+
+   private V compute(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction, Boolean computeIfPresent) {
```

Comment:
The `computeIfPresent` argument can't be null as far as I know. It seems that the code is mixing `Boolean` and `boolean` a lot on here. As far as I can see `boolean` should be used for all of these. Either way make sure they are consistent.